src/fixedpoint.ts

```diff
@@ -46,21 +111,40 @@ export class FixedPortion<T extends number> extends FixedInt<T> {
     this.max = max;
   }
 
+  /**
+   * Set the value of the fixed point number.
+   *
+   * Does not change the decimals.
+   *
+   * If the value is greater than the maximum value, this will throw an error.
+   */
   set val(value: bigint) {
     if (value > this.max) {
       throw new Error(`value ${value} is greater than max ${this.max}`);
     }
 
     this._value = value;
   }
+
   get val() {
     return this._value;
   }
 
+  /**
+   * Represent the fixed point number as a percentage value.
+   *
+   * If the underlying value is greater than Number.MAX_VALUE
+   * (1.7976931348623157e+308), this will return 'Infinity'.
+   *
+   * @returns The fixed point number as a percentage.
+   */
   formatPercentage(): number {
     return this.toFloat() * 100;
   }
 
+  /**
+   * TODO: this is a bit confusing as it implies it sets the percentage, which I would assume is based on the max value
+   */
   setPercentage(percentage: number): void {
```

Comment:
Shouldn't this take into account the max?

Reply:
Yeah it could/should. Im pretty sure it actually doesn't matter tho 😂 . E.g. reserved rate of 1_000 is still just 10%, in the context of it being a portion of 10_000.

I wonder if there really even needs to be a `max`. I wonder if there even needs to be this class, tbh.

Could we just have FixedInt have a setPercentage and formatPercentage that assumes a maximum of `100 parsed with x decimals`? Maybe?

Reply:
I think we should move to this being immutable if possible

If you think that there doesn't need to be a max, then yah looks nuke.

I think percentage should just be external functions we can call instead of class methods

wdyt?